pkg/cosmos/client.go

```diff
@@ -18,13 +18,26 @@ import (
 	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
 )
 
-// Client is a diagnostics client for DC/OS.
-type Client struct {
+// Client is a Cosmos client abstraction
+type Client interface {
```

Comment:
What is the reason for introducing this interface?

Reply:
This interface is used to inject mock inplementation in test.
@makkes requested this change https://github.com/dcos/dcos-core-cli/pull/360#discussion_r326988597

Reply:
I see, I had missed that as the file is hidden like other vendor files on Github.